### DISCUSSION TITLE
JV-880 Support - reading ROM patches?

### DISCUSSION THREAD

**Discussion by DrMechanica** [christofmuc/KnobKraft-orm#330] [Q&A]
I see the documentation lists support for the JV-880, but in the setup tab I only see the JV-80, which I understand is similar, but I'm not sure if the architecture is the same? I'm having trouble importing all of the patches off of my 880. It imports only 64 patches, but I know there are more on the device. I'm not sure if they're in a separate bank that I need to switch to, or something like that. If anyone can help me understand this better I'd appreciate it greatly, as I'd really like to be able to access all of the sounds on this device via knobkraft.

Thanks,
-M.

**Comment by christofmuc** [comment]
Very good question. We'll have to dig bit. When I did the JV-80 I only had a XV-3080 with backward compatibility mode.

My code for the JV-80 states: "# JV-80. The JV-880/JV-90/JV-1000 share the model ID of the JV-80, but have one respectively two bytes more size than the JV-80. How do I handle that?"

So when we use the current JV-80 adaptation, we'll probably be missing one parameter. Might or might not be important, but that's the first difference.

Secondly, I remember that the system can read out the patches and performances directly only for the memory bank, which are these 64 patches you can see. The additional patches in the device are ROM patches I think, and are imported via a different mode. Trying to remember this, it has been a while. Maybe we can only import them via the edit buffer.

Can you try to select a patch from one of the other banks and do a manual "get edit buffer from device" via the menu? If that works, we could use program change/bank change messages to import the remaning banks.

**Comment by christofmuc** [comment]
Maybe I am mistaken, and we can address at least the card directly as well. Here on page 223 it seems the card has another address:

http://cdn.roland.com/assets/media/pdf/JV-880_OM.pdf

on page 25 they have the full memory layout: The JV-880 has 2 banks of 64 patches in ROM (Preset). Additionally it has 64 user patches internally, and 64 more when you have inserted a card. The card can be addressed directly, we need to think how this can be implemented. The Preset patches probably can only be retrieved via the edit buffer as suggested above, which needs code changes, or to just copy them into the user memory and then retrieve them from there.

**Comment by DrMechanica** [comment]
Wow, thank you for the comprehensive and nigh immediate reply! So it seems that yes, importing the edit buffer does bring in the patches regardless of bank, allowing me to access them from KnobKraft.
